advisor-ui/src/pages/VolunteerHours/VolunteerHours.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useContext, useState } from "react";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import axios from "axios";
+import { UserContext } from "../../UserContext.js";
+import { IconButton } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { Link } from "react-router-dom";
+import "./VolunteerHours.css";
+
+export default function VolunteerHours() {
+  const [studentEvents, setStudentEvents] = useState([]);
+  const { user } = useContext(UserContext);
+
+  const fetchStudentEvents = async () => {
+    try {
+      const res = await axios.get(
+        `http://localhost:5000/api/student-event/${user.id}`,
+        { params: { studentId: user.id } }
+      );
+      const fetchedStudentEvents = res.data.studentEvents;
+      setStudentEvents(fetchedStudentEvents);
+    } catch (error) {
+      alert("Something went wrong!");
+    }
+  };
+
+  useEffect(() => {
+    fetchStudentEvents();
+  }, []);
+
+  const deleteStudentEventFromState = (id) => {
+    const filteredEvents = studentEvents.filter(
+      (studentEvent) => studentEvent.id !== id
+    );
+    setStudentEvents(filteredEvents);
+  };
+
+  const deleteStudentEventFromDB = async (id) => {
+    try {
+      const res = axios.delete(
+        `http://localhost:5000/api/student-event/${id}`,
+        { params: { studentEventId: id } }
+      );
+      return true;
+    } catch (error) {
+      alert("Unable to delete. Try again later.");
+      return false;
+    }
+  };
+
+  const handleDeleteStudentEvent = async (id) => {
+    // step 1 delete DB record
+    const successfulDelete = await deleteStudentEventFromDB(id);
+    // step 2 delete from local state copy
+    if (successfulDelete) {
+      deleteStudentEventFromState(id);
+    }
+  };
+
+  const createDateFromTimeStamp = (timestamp) => {
+    const date = new Date(timestamp);
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    return `${year}-${month}-${day}`;
+  };
```

Comment:
I think it'd be good to consider moving some functions to util files. This would be a good candidate for something like `dateTimeUtils.js` or something, because I wouldn't be surprised if you need to reuse it, or something similar (e.g. the inverse).

It's nice because you avoid duplicating logic across files if needed, but also it keeps all of your time related logic grouped up so if you need to make a change to the way you handle dates/time, you can do it all at once (e.g. moving from an older API to using locales).

I'd probably do the same with your CRUD functions; `studentEventUtils.js`, `studentUtils.js` etc; although you could combine them into a `DBUtils.js` for now as well since there's relatively few.